op-challenger/game/monitor.go

```diff
@@ -121,6 +128,9 @@ func (m *gameMonitor) progressGames(ctx context.Context, blockHash common.Hash,
 		}
 		gamesToPlay = append(gamesToPlay, game)
 	}
+	if err := m.claimer.Schedule(blockNumber, gamesToPlay); err != nil {
```

Comment:
this skips over games that aren't on the allowlist. Which could be problematic if one later deploys an op-challenger that's configured to whitelist an address that has not yet been claimed. (Which I can easily see happening as one tries to load balance op-challengers).

The issue gets more acute if there are multiple op-challengers using different senders (which ideally one should to avoid nonce conflicts). Since none of the other op-challenger deployments will ever be able to claim the bonds of others.

So for safety I recommend ignoring the allowlist for bond claims as it's inexpensive to act as a bond claimer. Also fine with leaving it as-is, but we should document this footgun somewhere so operators don't leave money on the table.

Reply:
It should be NBD to have the challenger perform the few extra static calls to check if it can claim bonds in those additional games not on the allowlist. bond claiming is entirely isolated to a new thread anyways.